account_financial_report/models/account_group.py

```diff
@@ -57,17 +57,9 @@ def _compute_level(self):
             else:
                 group.level = group.parent_id.level + 1
 
-    @api.depends(
-        "code_prefix_start",
-        "account_ids",
-        "account_ids.code",
-        "group_child_ids",
-        "group_child_ids.account_ids.code",
-    )
+    @api.depends("parent_path", "account_ids")
     def _compute_group_accounts(self):
-        account_obj = self.env["account.account"]
-        accounts = account_obj.search([])
-        for group in self:
-            prefix = group.code_prefix_start if group.code_prefix_start else group.name
-            gr_acc = accounts.filtered(lambda a: a.code.startswith(prefix)).ids
-            group.compute_account_ids = [(6, 0, gr_acc)]
+        for one in self:
+            one.compute_account_ids = one.account_ids | one.group_child_ids.account_ids
```

Comment:
If I understand correctly the purpose of the field, is to pre-compute all the accounts that belong this group or one of its descendants. As such, why not simply search for all the groups which `parent_path` contains the `parent_path` of this group, and join the `account_ids` of them?

If not, I see your code is not complete for 3 levels or more.

Reply:
Hmm true... But I wanted to avoid calling to `search()` to be able to leverage prefetching. Let me check if I can use the recursive solution you mentioned earlier.

Reply:
Thanks, fixed now. I tested it and it still gives almost the same performance as the original fix.